SCRIPTS: Downward Helix

Zenny89 opened the pull request:
Also some Erlene clean-up from earlier quests.

TeoTwawki commented on scripts/zones/Southern_San_dOria_[S]/Zone.lua:
```diff
@@ -28,6 +28,9 @@ function onZoneIn(player,prevZone)
     end
     if (prevZone == 81 and player:getQuestStatus(CRYSTAL_WAR, KNOT_QUITE_THERE) == QUEST_ACCEPTED and player:getVar("KnotQuiteThere") == 2) then
         cs = 0x003E;
+    end
```
Use an `elseif` instead.

TeoTwawki commented on scripts/zones/The_Eldieme_Necropolis_[S]/npcs/Erlene.lua:
```diff
@@ -21,16 +21,16 @@ require("scripts/globals/status");
     local ALittleKnowledge = player:getQuestStatus(CRYSTAL_WAR, A_LITTLE_KNOWLEDGE);
     local ALittleKnowledgeProgress = player:getVar("ALittleKnowledge");
-	
+    
```
Trim trailing.

TeoTwawki commented on scripts/zones/The_Eldieme_Necropolis_[S]/npcs/Erlene.lua:
```diff
@@ -21,16 +21,16 @@ require("scripts/globals/status");
                 player:getMainJob() == JOB_SMN or
                 player:getMainJob() == JOB_BLU) then
-                player:startEvent(12, 1);
+                player:startEvent(0x000C, 1);
```
Just so you know you do not have to change these to hex. I personalty prefer we un-hex-ify things that do not explicitly require it (or that do not combine with other values such as a mask) just because people who aren't hex savy get confused by it (dsp's entire wiki shows event IDs in decimal..).

TeoTwawki commented on scripts/zones/The_Eldieme_Necropolis_[S]/npcs/Erlene.lua:
```diff
@@ -48,40 +48,51 @@ function onTrigger(player,npc)
     local onSabbaticalProgress = player:getVar("OnSabbatical");
+    local downwardHelix = player:getQuestStatus(CRYSTAL_WAR, DOWNWARD_HELIX);
 
-    if (ENABLE_WOTG == 1 and ALittleKnowledge == QUEST_AVAILABLE) then
```
Please don't remove the settings without a good (and well explained) reason.

TeoTwawki replied:
ENABLE_WOTG exists with the other expansion controls so server operators can runs servers like it was 2004, or w/e.

Zenny89 commented:
Updated with some of the changes. I changed the values to hex because Demolish said they were more readable that way and that's what the Wiki has. I don't personally care either way, but it should be consistent. As for the "ENABLE_WOTG", I see no reason why there should be a check at this particular NPC. If a server is running like it's on 2004, there would be no reason for a player to even be in this zone. That behavior is handled in the Cavernous Maw files.

TeoTwawki commented:
@Zenny89  Its probably there so unlocking the job can be disabled by expansion even if you have `@zone` or any other means of entering the zone available (like a last ditch safety valve). But my main complaint is: *say what yer doin when ya do that*. The commit message just says `SCRIPTS: Downward Helix` and doesn't mention this other change (just FYI, I am not overly fond of catering to the *oldschool onry* crowd anyways).

You already know I am big on consistency so no problem with making the format of the startEvent() call consistent, I was just making sure you knew you could choose either one and it'd still work. A lot of times I see people thinking they have to be hex and thinking thats why some thing or another isn't working the way they expected.

@xdemolish why you havin people hexify things that don't need hexed :frowning: https://wiki.dspt.info/index.php/Northern_San_dOria_Event_IDs all decimal!  I happened to be de-hexing a lot of these as I edit. https://github.com/DarkstarProject/darkstar/commit/833b71543f47cf4b9a1348c891c41a91ef99a5b4#diff-d6329f79f59fea38f24f8da20084b7ebR49

xdemolish commented:
hex is easier on my eyes

TeoTwawki commented:
@xdemolish :worried: :interrobang:
